runtime/signals.c

```diff
@@ -329,14 +330,14 @@ CAMLprim value caml_install_signal_handler(value signal_number, value action)
     res = Val_int(1);
     break;
   case 2:                       /* was Signal_handle */
-    caml_read_field(caml_read_root(caml_signal_handlers), sig, &handler);
+    caml_read_field(caml_signal_handlers, sig, &handler);
     res = caml_alloc_1 (0, handler);
     break;
   default:                      /* error in caml_set_signal_action */
     caml_sys_error(NO_ARG);
   }
   if (Is_block(action)) {
-    caml_modify_field(caml_read_root(caml_signal_handlers), sig, Field(action, 0));
+    caml_modify_field(caml_signal_handlers, sig, Field(action, 0));
```

Comment:
Is this `caml_modify_field` correct? Isn't `caml_signal_handlers` registered as a generational global root that needs to be modified with `caml_modify_generational_global_root`?

Reply:
Yes, fixed it.